src/app/shell/details/details.component.ts

```diff
@@ -0,0 +1,127 @@
+import { delay, distinctUntilChanged, filter, takeUntil } from 'rxjs/operators';
+import { Component, OnInit, OnDestroy } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { Actions, ofAction, Select, Store } from '@ngxs/store';
+import { Observable, Subject } from 'rxjs';
+import { Workshop, WorkshopCard } from 'src/app/shared/models/workshop.model';
+import { NavBarName } from 'src/app/shared/enum/navigation-bar';
+import { AddNavPath, DeleteNavPath } from 'src/app/shared/store/navigation.actions';
+import { GetProviderById, GetWorkshopById, GetWorkshopsByProviderId, OnCreateRatingSuccess } from 'src/app/shared/store/user.actions';
+import { UserState } from 'src/app/shared/store/user.state';
+import { NavigationBarService } from 'src/app/shared/services/navigation-bar/navigation-bar.service';
+import { Provider } from 'src/app/shared/models/provider.model';
+import { RegistrationState } from 'src/app/shared/store/registration.state';
+import { GetRateByEntityId } from 'src/app/shared/store/meta-data.actions';
+import { AppState } from 'src/app/shared/store/app.state';
+import { EntityType } from 'src/app/shared/enum/role';
+
+@Component({
+  selector: 'app-details',
```

Comment:
not clear. we have app-details and app-details-page, little bit confusing. try to give descriptive names